### ISSUE TITLE
Add "typeswitch" linter

### ISSUE THREAD

**Issue by leonklingele** [golangci/golangci-lint#3487]
### Your feature request related to a problem? Please describe.

The `typeswitch` linter finds a type which implement an interfaces which are used in type-switch but the type does not appear in any cases of the type-switch:

```go
package main

type I interface{ F() }
type A struct{I} // implements I
type B struct{I} // implements I

func main() {
	var i I = A{}
	switch i.(type) {
	case A:
	}
}
```

```sh
$ go vet -vettool=`which typeswitch` main.go
./main.go:9:2: type B does not appear in any cases
```

https://github.com/gostaticanalysis/typeswitch

### Describe the solution you'd like.

—

### Describe alternatives you've considered.

—

**Comment by Antonboom** [comment]
Looks like contribution in https://golangci-lint.run/usage/linters/#exhaustive

**Comment by ldez** [comment]
This will produce a lot of false positives because interfaces are implicit.
